test/browser/fixture.test.js

```diff
@@ -0,0 +1,220 @@
+			expect(window.scrollY).to.equal(300);
+			await fixture(elem);
+			expect(window.scrollX).to.equal(0);
+			expect(window.scrollY).to.equal(0);
+		});
+
+		it('should reset mouse position', async() => {
+			const elem = html`<p>hello</p>`;
+			await fixture(elem);
+			await sendMouse({ type: 'move', position: [5, 10] });
+			expect(mousePos.x).to.equal(5);
+			expect(mousePos.y).to.equal(10);
+			await fixture(elem);
+			expect(mousePos.x).to.equal(0);
+			expect(mousePos.y).to.equal(0);
+		});
+
+		it('should reset focus', async() => {
+			const elem = await fixture(html`<button>hello</button>`);
+			expect(document.activeElement).to.equal(document.body);
+			await focusWithKeyboard(elem);
+			expect(document.activeElement).to.equal(elem);
+			await fixture(html`<div>hello</div>`);
+			expect(document.activeElement).to.equal(document.body);
+		});
+
+		it('should reset text direction', async() => {
+			await fixture(html`<p>hello</p>`);
+			expect(document.documentElement.hasAttribute('dir')).to.be.false;
+			await fixture(html`<p>mrhban</p>`, { rtl: true });
+			expect(document.documentElement.getAttribute('dir')).to.equal('rtl');
+			await fixture(html`<p>hello</p>`);
+			expect(document.documentElement.hasAttribute('dir')).to.be.false;
+		});
+
+		it('should use RTL text direction with AR language', async() => {
+			await fixture(html`<p>mrhban</p>`, { lang: 'ar' });
+			expect(document.documentElement.getAttribute('dir')).to.equal('rtl');
+		});
+
+		it('should default language to EN', async() => {
+			await fixture(html`<p>hello</p>`);
+			expect(document.documentElement.getAttribute('lang')).to.equal('en');
+		});
+
+		it('should reset language to EN', async() => {
+			await fixture(html`<p>bonjour</p>`, { lang: 'fr' });
+			expect(document.documentElement.getAttribute('lang')).to.equal('fr');
+			await fixture(html`<p>hello</p>`);
+			expect(document.documentElement.getAttribute('lang')).to.equal('en');
+		});
+
+		it('should default viewport size to 800x800', async() => {
+			await fixture(html`<p>hello</p>`);
+			expect(window.innerHeight).to.equal(800);
+			expect(window.innerWidth).to.equal(800);
+		});
+
+		it('should reset viewport size', async() => {
+			const elem = html`<p>hello</p>`;
+			await fixture(elem, { viewport: { height: 200, width: 300 } });
+			expect(window.innerHeight).to.equal(200);
+			expect(window.innerWidth).to.equal(300);
+			await fixture(elem);
+			expect(window.innerHeight).to.equal(800);
+			expect(window.innerWidth).to.equal(800);
+		});
+
+	});
+
+	describe('waitForElem', () => {
+
+		afterEach(() => {
+			resolves.clear();
+		});
+
+		it('should wait for element at fixture root', async() => {
+			const finishedPromise = fixture(`<${slowElem} id="elem1"></${slowElem}>`)
+				.then((elem) => elem.finished);
+			await waitUntil(() => resolves.has('elem1'));
+			const timeout = setTimeout(() => resolves.get('elem1')(), 50);
+			const finished = await finishedPromise;
+			clearTimeout(timeout);
```

Comment:
Why do we need to clear the timeout?

Reply:
It's not necessary when everything is working correctly, but when things start to go wrong -- like if `finishedPromise` resolves _before_ the code in the timeout executes -- then the test can fail (as it should) but then the code in the timeout executes during the NEXT test.

But explaining this out gave me an idea to clean them up in `afterEach` which should also make it clear why it's needed, so I'll make that change now!

Reply:
Awesome, yeah I like that much better!